File: python/sts_policy_test.py
#!/usr/bin/env python
# coding=utf-8
from sts import Sts, Scope


def test_policy():
    scope = Scope('name/cos:PutObject', 'test-1250000000', 'ap-guangzhou', 'dir/*')
    scopes = list()
    scopes.append(scope)
    print(Sts.get_policy(scopes=scopes))


def test_policy2():
    scopes = list()
    scopes.append(Scope('name/cos:PutObject', 'test-1250000000', 'ap-guangzhou', 'dir/*'))
    scopes.append(Scope('name/cos:GetObject', 'test-1250000000', 'ap-guangzhou', 'dir/*'))
    scopes.append(Scope('name/cos:InitiateMultipartUpload', 'test-1250000000', 'ap-guangzhou', 'dir/*'))
    scopes.append(Scope('name/cos:ListParts', 'test-1250000000', 'ap-guangzhou', 'dir/*'))
    scopes.append(Scope('name/cos:UploadPart', 'test-1250000000', 'ap-guangzhou', 'dir/*'))
    scopes.append(Scope('name/cos:CompleteMultipartUpload', 'test-1250000000', 'ap-guangzhou', 'dir/*'))
    scopes.append(Scope('name/cos:GetObject', 'test-1250000000', 'ap-guangzhou', '1/test.txt'))
    print(Sts.get_policy(scopes))


def test_sts():
    scopes = list()
    scopes.append(Scope('name/cos:PutObject', 'test-1250000000', 'ap-guangzhou', 'dir/*'))
    scopes.append(Scope('name/cos:GetObject', 'test-1250000000', 'ap-guangzhou', 'dir/*'))
    scopes.append(Scope('name/cos:InitiateMultipartUpload', 'test-1250000000', 'ap-guangzhou', 'dir/*'))
    scopes.append(Scope('name/cos:ListParts', 'test-1250000000', 'ap-guangzhou', 'dir/*'))
    scopes.append(Scope('name/cos:UploadPart', 'test-1250000000', 'ap-guangzhou', 'dir/*'))
    scopes.append(Scope('name/cos:CompleteMultipartUpload', 'test-1250000000', 'ap-guangzhou', 'dir/*'))
    scopes.append(Scope('name/cos:GetObject', 'test-1250000000', 'ap-guangzhou', '1/test.txt'))
    config = {
        # 临时密钥有效时长，单位是秒
        'duration_seconds': 1800,
        'secret_id': 'AKIDXXX',
        # 固定密钥
        'secret_key': 'EH8oXXX',
        'proxy': {
            'http': 'XXX',
            'https': 'XXX'
        },
        'policy': Sts.get_policy(scopes)
    }

    sts = Sts(config)
    response = sts.get_credential()
    print('get data : ' + str(response))


if __name__ == '__main__':
    test_policy()
    test_policy2()
    test_sts()

File: php/demos/sts_policy_test.php
<?php

require_once __DIR__ . '/../vendor/autoload.php';

$sts = new \Qcloud\STS\STS();
$scopes = array();
array_push(
    $scopes,
    new \Qcloud\STS\Scope(
        "name/cos:PutObject",
        "test-125000000",
        "ap-guangzhou",
        "/exampleobject"
    )
);
array_push(
    $scopes,
    new \Qcloud\STS\Scope(
        "name/cos:GetObject",
        "test-125000000",
        "ap-guangzhou",
        "/1.txt"
    )
);
$config = array(
    'url' => 'https://sts.tencentcloudapi.com/',
    'domain' => 'sts.tencentcloudapi.com',
    'proxy' => 'XX:XXX',
    'secretId' => 'AKIXXX', // 固定密钥
    'secretKey' => 'EH8XX', // 固定密钥
    'durationSeconds' => 1800, // 密钥有效期
    'policy' => $sts->getPolicy($scopes)
);
// 获取临时密钥，计算签名
try {
    $tempKeys = $sts->getTempKeys($config);
} catch (\Exception $e) {
    var_dump($e->getCode());
    var_dump($e->getMessage());
    exit;
}

echo json_encode($tempKeys);
